### PR TITLE
move helper functions to common

### PR DESCRIPTION
**Pull request by anara123**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Move helper functions to CommonServerPython
arg_to_int
arg_to_timestamp

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -2735,6 +2750,99 @@ def __init__(self, indicators=None, category=None, message=None):
...
+        throws exception if ``True`` and argument provided is None
+
+    :return:
+        returns an ``int`` if arg can be converted
+        returns ``None`` if arg is ``None`` and required is set to ``False``
+        otherwise throws an Exception
+    :rtype: ``Optional[int]``
+    """
+
+    if not arg:
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2781]: Won't this do problems is arg==0 ?

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -2735,6 +2750,99 @@ def __init__(self, indicators=None, category=None, message=None):
...
+        if required is True:
+            raise ValueError('Missing "{}"'.format(arg_name))
+        return None
+    if isinstance(arg, str):
+        if arg.isdigit():
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2786]: Maybe want to use `isdecimal`?

Comment (glicht) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2786]: Actually see that in py2 isdecimal is only for unicode. So ignore.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -2735,6 +2750,99 @@ def __init__(self, indicators=None, category=None, message=None):
...
+
+    This function is used to quickly validate an argument provided to XSOAR
+    via ``demisto.args()`` into an ``int`` type. It will throw a ValueError
+    if the input is invalid. If the input is None, it will throw a ValueError
+    if required is ``True``, or ``None`` if required is ``False.
+
+    :type arg: ``Any``
+    :param arg: argument to convert
+
+    :type arg_name: ``str``
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2767]: Why do we need `arg_name`? Just for printing the errors? It looks to me like something we can remove.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -28,6 +28,7 @@
     from requests.adapters import HTTPAdapter
     from urllib3.util import Retry
     from typing import Optional, List, Any
+    import dateparser
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:31]: demisto/python:1.3-alpine doesn't have the dateparser package. Need to do a conditional import if we want to use this.

**Comment by glicht** [comment]
we have failures in the build because of name conflicts. Maybe think of a different name or namespace these functions in to a a class with static functions.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 3 above)
Comment (anara123) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2767]: Yes, to print errors. The purpose of this code was to provide easy way for developers to assert and convert their args/params. 
Without returning arg name or param name, we lose the purpose.
But instead I can just print `aa is not a valid number` `2012-22-34 is not a valid date/time range` 
What do you think @glicht ?

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 4 above)
Comment (anara123) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:31]: It is already in try/except

Comment (glicht) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:33]: ok

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -28,6 +28,10 @@
     from requests.adapters import HTTPAdapter
     from urllib3.util import Retry
     from typing import Optional, List, Any
+
+    global dateparser
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:32]: Why this?

Comment (anara123) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:32]: will remove that. there was a unit test failing. Was trying something

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 3 above)
Comment (glicht) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2789]: Sounds good. Let's remove the arg_name or give it an empty value by default so this function can be used as a generic function.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 1 above)
Comment (glicht) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2803]: @anara123 see my comment

Comment (anara123) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:2803]: right, will change

**Review by glicht — APPROVED** [review]

**Pull request merged**